pkg/api/v1/router_bom_test.go

```diff
@@ -0,0 +1,206 @@
+	})
+
+	for _, s := range serialNumInfos {
+		sortAocMacAddrBoms(s.aocMacAddrsBom)
+		sortBmcMacAddrBoms(s.bmcMacAddrsBom)
+	}
+}
+
+func sortAocMacAddrBoms(aocMacAddrBoms []AocMacAddressBom) {
+	sort.Slice(aocMacAddrBoms, func(i, j int) bool {
+		return aocMacAddrBoms[i].AocMacAddress < aocMacAddrBoms[j].AocMacAddress
+	})
+}
+
+func sortBmcMacAddrBoms(bmcMacAddrBoms []BmcMacAddressBom) {
+	sort.Slice(bmcMacAddrBoms, func(i, j int) bool {
+		return bmcMacAddrBoms[i].BmcMacAddress < bmcMacAddrBoms[j].BmcMacAddress
+	})
+}
+
+func TestParseXlsxFile(t *testing.T) {
+	var testCases = []struct {
+		testName                 string
+		filePath                 string
+		expectedErr              bool
+		expectedErrMsg           string
+		expectedSerialNumBomInfo []*SerialNumBomInfo
+	}{
+		{
+			testName:                 "file missing serial number",
+			filePath:                 "./testdata/test_empty_serial.xlsx",
+			expectedErr:              true,
+			expectedErrMsg:           "empty serial number",
+			expectedSerialNumBomInfo: nil,
+		},
+		{
+			testName:                 "file missing aocMacAddress ",
+			filePath:                 "./testdata/test_empty_aocMacAddress.xlsx",
+			expectedErr:              true,
+			expectedErrMsg:           "empty aoc mac address",
+			expectedSerialNumBomInfo: nil,
+		},
+		{
+			testName:                 "file missing bmcMacAddress",
+			filePath:                 "./testdata/test_empty_bmcMacAddress.xlsx",
+			expectedErr:              true,
+			expectedErrMsg:           "empty bmc mac address",
+			expectedSerialNumBomInfo: nil,
+		},
+		{
+			testName:                 "valid file for single bom",
+			filePath:                 "./testdata/test_valid_one_bom.xlsx",
+			expectedErr:              false,
+			expectedErrMsg:           "",
+			expectedSerialNumBomInfo: []*SerialNumBomInfo{testSerialNumBomInfo1},
+		},
+		{
+			testName:                 "valid file for multiple bom",
+			filePath:                 "./testdata/test_valid_multiple_boms.xlsx",
+			expectedErr:              false,
+			expectedErrMsg:           "",
+			expectedSerialNumBomInfo: []*SerialNumBomInfo{testSerialNumBomInfo1, testSerialNumBomInfo2},
+		},
+		{
+			testName:                 "file missing SERIALNUM col",
+			filePath:                 "./testdata/test_empty_serial_col.xlsx",
+			expectedErr:              true,
+			expectedErrMsg:           "missing colomn, serial num -1, sub-item 4, sub-serial 5",
+			expectedSerialNumBomInfo: nil,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testName, func(t *testing.T) {
+			file, err := os.Open(tt.filePath)
+			if err != nil {
+				t.Fatalf("os.Open(%v) failed to open file %v\n", tt.filePath, err)
+			}
+
+			// Translate file to bytes since ParseXlsxFile accepts bytes of file as argument,
+			// which is the format of a file reading from the HTTP request.
+			stat, err := file.Stat()
+			if err != nil {
+				fmt.Println(err)
```

Comment:
you have a couple of better (IMO) options here. Consider `t.Logf`, `t.Fatalf` (better, I think) or `require.NoError` (if you are familiar with `github.com/stretchr/testify/require`.